docker/install-dynamic-plugins.py

```
@@ -789,6 +792,87 @@ def wait_for_lock_release(lock_file_path):
      time.sleep(1)
    print("======= Lock released.")
 
+def extract_catalog_index(catalog_index_image: str, catalog_index_mount: str) -> str:
+    """Extract the catalog index OCI image and return the path to dynamic-plugins.default.yaml if found."""
+    if not catalog_index_image:
+        print("======= No CATALOG_INDEX_IMAGE specified, skipping catalog index extraction", flush=True)
+        return None
+    try:
+        print(f"\n======= Extracting catalog index from {catalog_index_image}", flush=True)
+        skopeo_path = shutil.which('skopeo')
+        if skopeo_path is None:
+            print("WARNING: skopeo executable not found in PATH, skipping catalog index extraction", flush=True)
+            return None
+
+        catalog_index_temp_dir = os.path.join(catalog_index_mount, '.catalog-index-temp')
+        os.makedirs(catalog_index_temp_dir, exist_ok=True)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            if not catalog_index_image.startswith('docker://'):
+                image_url = f'docker://{catalog_index_image}'
+            else:
+                image_url = catalog_index_image
+            print(f"\t==> Copying catalog index image to local filesystem", flush=True)
+            local_dir = os.path.join(tmp_dir, 'catalog-index-oci')
+
+            # Download the OCI image using skopeo
+            result = subprocess.run(
+                [skopeo_path, 'copy', image_url, f'dir:{local_dir}'],
+                capture_output=True,
+                text=True
+            )
+            if result.returncode != 0:
+                print(f"WARNING: Failed to download catalog index image: {result.stderr}", flush=True)
+                return None
```

Same point: it should raise an exception.